pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go

```diff
@@ -428,14 +437,48 @@ func (cra *cpuResourceAdvisor) assembleProvision() (InternalCalculationResult, e
 
 	// Fill in reclaimed pool size of non binding numas
 	reservePoolSizeOfNonBindingNumas := int(math.Ceil(float64(reservePoolSize*cra.nonBindingNumas.Size()) / float64(cra.metaServer.NumNUMANodes)))
-	reclaimPoolSizeOfNonBindingNumas := cra.nonBindingNumas.Size()*cra.metaServer.CPUsPerNuma() - nonNumaBindingRequirement - reservePoolSizeOfNonBindingNumas
+
+	reclaimPoolSizeOfNonBindingNumas := general.Max(cra.nonBindingNumas.Size()*cra.metaServer.CPUsPerNuma()-nonNumaBindingRequirement-reservePoolSizeOfNonBindingNumas,
+		int(math.Ceil(float64(minReclaimCPURequirement)/float64(cra.metaServer.NumNUMANodes)))*cra.nonBindingNumas.Size())
+	sharePoolSize := cra.nonBindingNumas.Size()*cra.metaServer.CPUsPerNuma() - reclaimPoolSizeOfNonBindingNumas - reservePoolSizeOfNonBindingNumas
+
+	sharePools := genShareRegionPools(shareRegionRequirement, sharePoolSize)
+	for poolName, size := range sharePools {
+		provision.PoolEntries[poolName] = make(map[int]resource.Quantity)
+		provision.PoolEntries[poolName][cpuadvisor.FakedNumaID] = *resource.NewQuantity(int64(size), resource.DecimalSI)
+	}
+
 	if provision.PoolEntries[state.PoolNameReclaim] == nil {
 		provision.PoolEntries[state.PoolNameReclaim] = make(map[int]resource.Quantity)
 	}
 	provision.PoolEntries[state.PoolNameReclaim][cpuadvisor.FakedNumaID] = *resource.NewQuantity(int64(reclaimPoolSizeOfNonBindingNumas), resource.DecimalSI)
 	return provision, nil
 }
 
+func genShareRegionPools(shareRegionRequirement map[string]int, sharePoolSize int) map[string]int {
```

Comment:
maybe this function should be moved below to make sure the method for cpuResourceAdvisor place together, just for a better coding style

Reply:
and it would be better if more comments are added for this function